nulib/meadow
# Use externally managed secrets for Meadow configuration

# Summary 

Refactor Meadow's configuration system to use externally managed secrets via AWS Secrets Manager. This is a work in progress. Mergeable as is, but there are still some more bits of configuration to source from elsewhere. I have not yet terraformed the necessary changes to staging, because it will break the existing staging deploy if done before merging.

# Specific Changes in this PR

- Move all dynamic secrets to `Meadow.Config.Runtime` namespace
- Remove janky `Env` module and `Hush` dependency
- Remove Hush from test_helper.exs
- Consolidate component configurations into compile-time/runtime
- Push test configs into localstack secrets manager
- Turn off warnings on lambda logging

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

Run Meadow in your dev environment as usual and try to check as much functionality as possible.

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
